examples/weather/unified_recipe/train.py

```diff
@@ -0,0 +1,358 @@
+        }
+    )
+    model = model.to(dist.device)
+
+    # Distributed learning
+    if dist.world_size > 1:
+        ddps = torch.cuda.Stream()
+        with torch.cuda.stream(ddps):
+            model = DistributedDataParallel(
+                model,
+                device_ids=[dist.local_rank],
+                output_device=dist.device,
+                broadcast_buffers=dist.broadcast_buffers,
+                find_unused_parameters=dist.find_unused_parameters,
+            )
+        torch.cuda.current_stream().wait_stream(ddps)
+        torch.device.synchronize()
+
+    # Initialize optimizer
+    OptimizerClass = getattr(optimizers, cfg.training.optimizer.name)
+    optimizer = OptimizerClass(model.parameters(), **cfg.training.optimizer.args)
+
+    # Attempt to load latest checkpoint if one exists
+    if dist.world_size > 1:
+        torch.distributed.barrier()
+    loaded_epoch = load_checkpoint(
+        "./checkpoints",
+        models=model,
+        optimizer=optimizer,
+        scheduler=None,
+        device=dist.device,
+    )
+
+    # Initialize filesytem
+    if cfg.filesystem.type == 'file':
+        fs = fsspec.filesystem(cfg.filesystem.type)
+    elif cfg.filesystem.type == 's3':
+        fs = fsspec.filesystem(cfg.filesystem.type,
+                               key=cfg.filesystem.key,
+                               secret=os.environ["AWS_SECRET_ACCESS_KEY"], 
+                               client_kwargs={'endpoint_url': cfg.filesystem.endpoint_url,
+                                              'region_name': cfg.filesystem.region_name})
+    else:
+        raise NotImplementedError(f'Filesystem type {cfg.filesystem.type} not implemented')
+
+    # Get filesystem mapper for datasets
+    train_dataset_mapper = fs.get_mapper(cfg.dataset.train_dataset_filename)
+    val_dataset_mapper = fs.get_mapper(cfg.dataset.val_dataset_filename)
+
+    # Initialize validation datapipe
+    val_datapipe = SeqZarrDatapipe(
+        zarr_dataset=zarr.open(val_dataset_mapper, mode='r'),
+        variables=['time', 'predicted', 'unpredicted'],
+        batch_size=cfg.validation.batch_size,
+        num_steps=cfg.validation.num_steps + cfg.training.nr_input_steps,
+        shuffle=False,
+        device=dist.device,
+        process_rank=dist.rank,
+        world_size=dist.world_size,
+    )
+
+    # Normalizer (TODO: Maybe wrap this into model)
+    predicted_batch_norm = nn.BatchNorm2d(cfg.dataset.nr_predicted_variables, momentum=None, affine=False).to(dist.device)
+    unpredicted_batch_norm = nn.BatchNorm2d(cfg.dataset.nr_unpredicted_variables, momentum=None, affine=False).to(dist.device)
+    def normalize_variables(variables, batch_norm):
+        shape = variables.shape
+        variables = variables.flatten(0, 1)
+        variables = batch_norm(variables)
+        variables = variables.view(shape)
+        return variables
```

Comment:
Should this be in the unified recipe? Why or why not?

Reply:
I am strongly in favor of these being in the unified recipe although there might be a better way to add them. I am adding some parts now where it saves the mean and std from these normalizes to a npy file that will be compatible with downstream tasks.

Reply:
We would expect that the mean and std computed from these `BatchNorm2d` operators will tend toward a sample mean and std, but what is the effect on the initial iterations (especially if you are limited to batch size of 1 like graphcast)?